lib/jnpr/junos/factcache.py

```diff
@@ -0,0 +1,280 @@
+from abc import ABCMeta
+import collections
+import warnings
+
+import jnpr.junos.facts
+import jnpr.junos.exception
+
+
+class _FactCache(collections.MutableMapping):
+    """
+    A dictionary-like object which performs on-demand fact gathering.
+
+    This class should not be used directly. An instance of this class is
+    available as the :attr:`facts` attribute of a Device object.
+
+    **Dictionary magic methods:**
+      * :meth:`__getitem__`: Gets the value of a given key in the dict.
+      * :meth:`__delitem__`: Called when a key is deleted from the dict.
+      * :meth:`__setitem__`: Called when a key is set on the dict.
+      * :meth:`__iter__`: Called when iterating over the keys of the dict.
+      * :meth:`__len__`: Called when getting the length of the dict.
+      * :meth:`__repr__`: Called when representing the dict as a string.
+
+    **Additional methods:**
+      * :meth:`_refresh`: Refreshes the fact cache.
+    """
+    # Used to register the class as a dict.
+    __metaclass__ = ABCMeta
```

Comment:
You are correct that this was not working as expected. The intention was to make the `_FactCache` class appear as a subclass of `dict`, but this was not working as expected and has been removed.